### PR TITLE
signmessage support

### PR DESCRIPTION
**Pull request by rustyrussell**
Special thanks to Twitter for supplying testcases:

https://twitter.com/rusty_twit/status/1182524014440222720

### CODE REVIEW

File: tests/test_misc.py [hunk 1]
```diff
@@ -1646,3 +1646,39 @@ def test_relative_config_dir(node_factory):
     assert os.path.isabs(l1.rpc.listconfigs()["lightning-dir"])
     l1.stop()
     os.chdir(initial_dir)
+
+
+def test_signmessage(node_factory):
+    l1 = node_factory.get_node()
+
+    corpus = [ [ None,
+                 "this is a test!",
```
Comment (trueptolemy) [inline tests/test_misc.py]: These `whitespace`s make travis-ci unhappy :-)

**Comment by rustyrussell** [comment]
> BTW, can lightning-checkmessage upgrade to json command from a tool in the future?

I thought about making it a command, but it really doesn't need a running node at all. A plugin would be trivial, however.

**Comment by niftynei** [comment]
strongly disagree with not providing a 'symmetric' decryption command with this. from the average user's perspective, the tool is not discoverable.

a plugin is fine, but we shouldn't ship this without an equally obvious, integrated lightning-cli command for decryption.

**Comment by darosior** [comment]
> a plugin is fine, but we shouldn't ship this without an equally obvious, integrated lightning-cli command for decryption.

On the same subject I began to make a `hsmtools` plugin to which I wanted to include `hsm_secret` decryption, and these two functionalities (so both the encryption and the decryption could be part of the same interface). We could also add https://github.com/ElementsProject/lightning/issues/3118 and even wallet locking (`wallettools`?).

**Comment by trueptolemy** [comment]
> On the same subject I began to make a `hsmtools` plugin to which I wanted to include `hsm_secret` decryption, and these two functionalities (so both the encryption and the decryption could be part of the same interface). We could also add #3118 and even wallet locking (`wallettools`?).

`hsmtools` and `wallettools` can be merged into one plugin to provides general encryption/decryption related functions?

**Comment by darosior** [comment]
> hsmtools and wallettools can be merged into one plugin to provides general encryption/decryption related functions?

Yeah that's what I propose

**Comment by rustyrussell** [comment]
Yes, a hsmtools combo plugin makes sense.

I'll move the standalone tool to devtools/ and move implementation into the core.

**Comment by rustyrussell** [comment]
Trivial rebase (CHANGELOG.md conflict).  All feedback addressed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -15,7 +15,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - JSON API: `listfunds` now lists a blockheight for confirmed transactions, and has `connected` and `state` fields for channels, like `listpeers`.
 - JSON API: `fundchannel_start` now includes field `scriptpubkey`
 - JSON API: New method `listtransactions`
-- JSON API: `signmessage` will now create a signature from your node on a message.
+- JSON API: `signmessage` will now create a signature from your node on a message; `checkmessage` will verify it.
```
Comment (niftynei) [inline CHANGELOG.md]: :+1: thanks!!

File: doc/lightning-checkmessage.7.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+The **checkmessage** RPC command is the counterpart to
+**signmessage**: given a node id (*pubkey*), signature (*zbase*) and a
+*message*, it verifies that the signature was generated by that node
+for that message (more technically: by someone who knows that node's
+secret).
+
+As a special case, if *pubkey* is not specified, we will try every
+known node key (as per *listnodes*), and verification succeeds if it
+matches for any one of them.  Note: this is implemented far more
```
Comment (niftynei) [inline doc/lightning-checkmessage.7.md]: @niftynei is laughing at 'implemented more efficiently'

**Comment by niftynei** [comment]
ACK https://github.com/ElementsProject/lightning/pull/3150/commits/4a73962e8aa0ac2f34b957ad36308a60975d6603

**Pull request merged**